x-pack/platform/plugins/shared/ml/public/locator/ml_management_locator.ts

```diff
@@ -83,11 +84,14 @@ export class MlManagementLocatorInternal {
     return path;
   };
 
-  public readonly getUrl = async (params: MlLocatorParams, appId: string = 'anomaly_detection') => {
-    const path = this.getPath(params);
+  public readonly getUrl = async (
+    params: MlLocatorParams | undefined,
+    appId: string = 'anomaly_detection'
+  ) => {
+    const path = params ? this.getPath(params) : '';
     const url = await this._locator?.getUrl({
       sectionId: this._sectionId,
-      appId: `${appId}/${path}`,
+      appId: `${appId}${path ? '/' + path : ''}`,
```

Comment:
@qn895 - could you give me some context on why this change was needed? Thanks!

Reply:
As discussed in Slack, this is due to the the url needing the slash previously. Now we the change of the paths we don't need it anymore.